andela/ah-robotics-frontend
# #162949042 Users should be able to comment on articles

#### What does this PR do?
Implement frontend functionality for commenting on articles
#### Description of Task to be completed?
- [x] setup actions, reducers, sagas
- [x] implement add comment feature
- [x] implement get all comments feature
- [x] implement update comment feature
- [x] implement delete comment feature

1. Navigate to a folder where you would prefer to test the repository.

2. Create a directory
`mkdir myproject`

3. Navigate to the directory "myproject"
`cd myproject`

4. Clone the author's haven repository
`git clone https://github.com/andela/ah-robotics-frontend.git`

5. Change the directory into the "ah-robotics-fronted"
`cd ah-robotics-frontend`

6.  Checkout to the branch ft-comments-crud-162949042
`git checkout ft-comments-crud-162949042`

7. Install dependencies from the package.json
 `npm install` to install all the modules listed in the `package.json` file.

8. Start the application in development mode
`npm start`
> The application opens http://localhost:3000 in the browser successfully if there are no issues.

 #### Test out the feature
The application can be accessed on heroku using the URL below : 
[https://ah-robotics-frontend-sta-pr-15.herokuapp.com](https://ah-robotics-frontend-sta-pr-15.herokuapp.com);

9. Register a new user by heading over to the **register** page.
10. After Registering a user, head over to the **login** page and use the details that you registered with to login

11. Select any article from the articles list in order to load the article view and access the comment section of the article.

12. Type a comment in the text area and then press the **Add Comment** button 
13. When you add a comment you can also **_Update_** it or **_Delete_** it as the options will appear below the comment text.

###  Running Tests
14. Add a **.env** file to the root folder of the project and add the configurations for social authentication in the following format, this is necessary for social login tests to pass.

           REACT_APP_API_KEY=sample_key
           REACT_APP_AUTH_DOMAIN=sample_domain
> **NOTE** : The values of the above variables are in the pinned messages section on the slack group.
15. Run the following command to make sure all tests are passing:
`npm test`

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?

[#162949042](https://www.pivotaltracker.com/story/show/162949042)

### Screenshots
1. Adding a comment
![image](https://user-images.githubusercontent.com/10683903/53671034-d13cf880-3c8d-11e9-94a3-67de96b524d2.png)

2. Updating a comment
![image](https://user-images.githubusercontent.com/10683903/53671108-119c7680-3c8e-11e9-8db5-53a17062f203.png)

3. Deleting a comment
![image](https://user-images.githubusercontent.com/10683903/53671124-2416b000-3c8e-11e9-834d-38ac6945c192.png)
